infra/data_stores/data_lake_stack.py

```diff
@@ -52,10 +52,39 @@ def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
         # Lambda Handler Functions
         dataset_handler_function = aws_lambda.Function(
             self,
-            "dataset-handler-function",
+            "datasets-endpoint-function",
+            function_name="datasets-endpoint",
             handler="function.lambda_handler",
             runtime=aws_lambda.Runtime.PYTHON_3_6,
-            code=aws_lambda.Code.from_asset(path="lambda/dataset-handler"),
+            code=aws_lambda.Code.from_asset(
+                path="lambda/datasets-endpoint",
+                bundling=core.BundlingOptions(
+                    image=aws_lambda.Runtime.PYTHON_3_6.bundling_docker_image,  # pylint:disable=no-member
+                    command=[
+                        "bash",
+                        "-c",
+                        "pip install -r requirements.txt -t /asset-output && cp -au . /asset-output",
+                    ],
+                ),
+            ),
+        )
+        db_datasets_table.add_global_secondary_index(
+            index_name="datasets_title",
+            partition_key=aws_dynamodb.Attribute(
+                name="sk", type=aws_dynamodb.AttributeType("STRING")
```

Comment:
`sk` means sort key. It is universal field which will be populated by different values types. I found this field name quite common in DynamoDB schema designs.

Reply:
`aws_dynamodb.AttributeType("STRING")` fixed in 6c2cbedb451a36b9889ae89ef3fcd08454c044c0 . Thanks.

Reply:
Can you please rebase this and the other fixes into the relevant commits? It'll make the history much easier to follow.

Reply:
> `sk` means sort key. It is universal field which will be populated by different values types. I found this field name quite common in DynamoDB schema designs.

Thanks, this is something I'll have to get used to. But is there some name which would be more descriptive of what is actually in this field? If not, "sort_key" would be easier to understand since it's searchable.

Reply:
I have a feeling that we somehow agreed to leave this as is yesterday and you pressed "Resolve Conversation" button
.

Reply:
Yeah, weird